scipy/sparse/_csr.py

```diff
@@ -122,6 +122,47 @@ def tobsr(self, blocksize=None, copy=True):
 
     tobsr.__doc__ = _spbase.tobsr.__doc__
 
+
+    def _broadcast_to(self, shape, copy=False):
+
+        old_shape = self.shape
+        
+        if old_shape == shape:
+            return self.copy() if copy else self
+        
+        if len(shape) != 2:
```

Comment:
Let's use the `sputil.check_shape()` function here, which also helpfully supports the 1d case if this is a csr_array.

Reply:
Done at a41da4b0720c0c1f44278fffa1ed133cbae9cb76.

Reply:
Does the newly added `check_shape` call need `allow_nd=self._allow_nd` parameter?

Reply:
Added it!